### ISSUE TITLE
Alternative zu libsodium

### ISSUE THREAD

**Issue by kikmedia** [richardhj/contao-newsletter2go-sync#5]
Moin Richard

Soweit ich sehen kann hast Du in https://github.com/richardhj/contao-newsletter2go-sync/commit/381473f2694304f00456f810348d942c423bacef den Support für mcrypt gedroppt und ausschliesslich auf libsodium gesetzt. So sehr ich moderne Crypto auch begrüsse: Es gibt leider etliche Hoster, die das zur Zeit noch blockieren, libsodium als 'Nice to have deklarieren und sich rundheraus weigern, es mit einzukompilieren. 

Beispiele: Hosteurope ("Wir haben keine Pläne"), Mittwald ("In unserem Standard-Linux-Stack ist das nicht integriert").

Dadurch ist leider die Extension dann nicht nutzbar, weil keine Keys gespeichert werden können. Magst Du das bitte noch einmal überdenken bzw. mcrypt als Fallback anbieten? Ich weiss, das ist nicht schön, aber so wie ich das sehe dürfte libsodium erst nach und nach verfügbar sein.

**Comment by richardhj** [comment]
Auf mcrypt als Fallback zu setzen heißt, auch auf PHP 7.1. zu bleiben bei dem jeweiligen Projekt. Wäre das eine Option?
Sonst halt openssl_encrypt…

**Comment by kikmedia** [comment]
Ja, das wäre eine Option, die aus meiner Sicht weitgehend failsafe ist. Mir ist klar, dass PHP 7.1 so gut wie tot ist. Openssl_encrypt ist sicherlich schöner, da gebe ich Dir Recht. Wenn Du generell PHP 7.2 als Voraussetzung hast, dann ist das doch auch okay.
